prov/verbs/src/verbs_msg.c

```diff
@@ -127,6 +211,21 @@ fi_ibv_msg_ep_sendmsg(struct fid_ep *ep_fid, const struct fi_msg *msg, uint64_t
 }
 
 static ssize_t
+fi_ibv_sstream_ep_sendmsg(struct fid_ep *ep_fid, const struct fi_msg *msg, uint64_t flags)
+{
+	ssize_t ret;
+
+	ret = fi_ibv_msg_ep_sendmsg(ep_fid, msg, flags);
+	if (ret)
+		return ret;
+
+	if ((ret = fi_ibv_get_send_comp(ep_fid)) == 0)
+		return fi_ibv_msg_total(msg);
```

Comment:
This semantic isn't going to work.  Verbs requires that the send and receive buffers be registered before being used.  There's no requirement for registration for the user of a stream EP.  The data will need to be copied into a pre-registered buffer prior to sending.

Additionally, blocking every send operation until a completion has been posted will result in horrible performance.  Completions are written by verbs hardware until the data has been acked by the target.  That's a significant delay.  We want to emulate nonblocking sockets, since that's how most applications use them.

Reply:
I see, so what would a non-blocking usage of SOCK_STREAM_EP look like?